Libraries/LibWeb/HTML/ImageData.cpp

```diff
@@ -163,4 +179,56 @@ const JS::Uint8ClampedArray* ImageData::data() const
     return m_data;
 }
 
+WebIDL::ExceptionOr<void> ImageData::serialization_steps(SerializationRecord& serialized, bool for_storage, SerializationMemory& memory)
+{
+    auto& vm = this->vm();
+
+    // 1. Set serialized.[[Data]] to the sub-serialization of the value of value's data attribute.
+    auto serialized_data = TRY(structured_serialize_internal(vm, m_data, for_storage, memory));
+    serialized.extend(move(serialized_data));
+
+    // 2. Set serialized.[[Width]] to the value of value's width attribute.
+    serialize_primitive_type(serialized, m_bitmap->width());
+
+    // 3. Set serialized.[[Height]] to the value of value's height attribute.
+    serialize_primitive_type(serialized, m_bitmap->height());
+
+    // 4. Set serialized.[[ColorSpace]] to the value of value's colorSpace attribute.
+    serialize_primitive_type(serialized, static_cast<u32>(m_color_space));
```

Comment:
I suppose this ends up being the same thing, but I think it would. be better to use serialize_enum here, I was a little confused for a second since I had an inkling color space would have `u8` as the underlying type, and similarly the serialization could take a `PredefinedColorSpace` as template param instead of u32.

(And this is super minor, but I suppose if we really wanted to avoid non-fixed size types which is generally a good thing for sending stuff over the wire, the width/height could be `WebIDL::UnsignedLong` since that matches the IDL, end though bitmap returns an int)

Reply:
Took me few days to come back to this. Fixed up so it uses `serialize_enum()` and changed the IDL exposed attributes to be `WebIDL::UnsignedLong`.  Also added spec links, couldn't for the life of me find those spec links when I was browsing the spec previously 😆

Reply:
Ah now I think there is a mismatch between int for serializing but deserization is int for the width/height

Reply:
@shannonbooth since the underlying type is `int` I think it is better to just serialize and deserialize them as the underlying type. Let me know if you disagree.

Reply:
Done